internal/resolver/delegatingresolver/delegatingresolver.go

```diff
@@ -65,15 +62,15 @@ func mapAddress(address string) (*url.URL, error) {
 			Host:   address,
 		},
 	}
-	url, err := httpProxyFromEnvironment(req)
+	url, err := HttpsProxyFromEnvironment(req)
 	if err != nil {
 		return nil, err
 	}
 	return url, nil
 }
 
-// New creates a new delegating resolver that will call the target and proxy
-// child resolvers based on the proxy enviorinment configurations.
+// New creates a new delegating resolver that which is used to call the target
+// and proxy child resolvers based on the proxy enviorinment configurations.
```

Comment:
> based on the proxy enviorinment configurations

this is not clear from the input. It can be changed to "It calls proxy child resolver if target endpoint points to a proxy"

Reply:
Done.